src/views/Generator/Generator.hooks.ts

```diff
@@ -89,3 +90,39 @@ export function useIsGeneratorDirty(fileName: string) {
     JSON.stringify(generatorStateData) !== JSON.stringify(generatorFileData)
   )
 }
+
+export function useScriptExport(generatorFileName: string) {
+  const showToast = useToast()
+  const setScriptName = useGeneratorStore((store) => store.setScriptName)
+  const { mutateAsync: updateGeneratorFile } =
+    useUpdateValueInGeneratorFile(generatorFileName)
+
+  return useCallback(
+    async (scriptName: string) => {
+      setScriptName(scriptName)
```

Comment:
I set the script name before doing anything else so that if an error occurs, the script name the user entered will still be used the next time they open the dialog.